var/spack/repos/builtin/packages/umt/package.py

```diff
@@ -0,0 +1,196 @@
+
+    # Override the parent class function until its fixed in spack.  (Greg Becker and Chris White are aware of bug)
+    def flag_handler(self, name, flags):
+        if name in ("cflags", "cxxflags", "cppflags", "fflags", "ldflags"):
+            return (None, None, None)  # handled in the cmake cache
+        return (flags, None, None)
+
+    def initconfig_compiler_entries(self):
+        spec = self.spec
+
+        if "umpire" in spec:
+            if "+rocm" in spec["umpire"]:
+                spec.compiler_flags["cxxflags"].append("-I{0}/include".format(spec["hip"].prefix))
+
+        #######################
+        # Note - call the super class AFTER changing any flags, as the super class
+        # adds the cflags, cxxflags, fflags, ldflags, etc, to the cache entries list.
+        # If you try adding any of these yourself you will end up with duplicates.
+        # - black27
+        entries = super(Umt, self).initconfig_compiler_entries()
+
+        if "+openmp" in spec:
+            if spec.satisfies("%xl"):
+                entries.append(cmake_cache_string("TETON_OpenMP_Fortran_FLAGS_RELEASE", "-qsmp=omp"))
+                entries.append(cmake_cache_string("TETON_OpenMP_Fortran_FLAGS_DEBUG", "-qsmp=omp"))
+                entries.append(cmake_cache_string("TETON_OpenMP_CXX_FLAGS_RELEASE", "-qsmp=omp"))
+                entries.append(cmake_cache_string("TETON_OpenMP_CXX_FLAGS_DEBUG", "-qsmp=omp"))
+
+                if "+openmp" in spec:
+                    entries.append(cmake_cache_string("TETON_OpenMP_Offload_Fortran_FLAGS", "-qoffload -qtgtarch=sm_70"))
+            else:
+                entries.append(cmake_cache_string("TETON_OpenMP_Fortran_FLAGS_RELEASE", "-fopenmp"))
+                entries.append(cmake_cache_string("TETON_OpenMP_Fortran_FLAGS_DEBUG", "-fopenmp"))
+                entries.append(cmake_cache_string("TETON_OpenMP_CXX_FLAGS_RELEASE", "-fopenmp"))
+                entries.append(cmake_cache_string("TETON_OpenMP_CXX_FLAGS_DEBUG", "-fopenmp"))
+
+        if spec.satisfies("%cce"):
+            entries.append(cmake_cache_option("STRICT_FPP_MODE", True))
+            entries.append(cmake_cache_string("TETON_OpenMP_Offload_LINK_FLAGS", "-ldl"))
+
+        return entries
+
+    def initconfig_hardware_entries(self):
+        spec = self.spec
+        entries = super(Umt, self).initconfig_hardware_entries()
+
+        #######################
+        # Parallelism
+        #######################
+        if "+cuda" in spec:
+            entries.append(cmake_cache_option("ENABLE_CUDA", True))
+        if "+openmp" in spec:
+            entries.append(cmake_cache_option("ENABLE_OPENMP", True))
+        if "+openmp_offload" in spec:
+            entries.append(cmake_cache_option("ENABLE_OPENMP_OFFLOAD", True))
+
+        if spec.satisfies("%cce"):
+            entries.append(cmake_cache_option("OPENMP_HAS_USE_DEVICE_ADDR", True))
+            entries.append(cmake_cache_option("OPENMP_HAS_FORTRAN_INTERFACE", True))
```

Comment:
This should probably be buried in the `if "+openmp" in spec` block above, and should probably be in the compilers section